internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixInstanceConfigRepository.java

```diff
@@ -138,4 +143,24 @@ public synchronized void onInstanceConfigChange(List<InstanceConfig> instanceCon
     LOGGER.info("New instance group id mapping: {}.", instanceGroupIdMapping);
     LOGGER.info("The total number of groups: {}.", groupCount);
   }
+
+  /**
+   * This function will assign a unique group id per group.
+   * If there is no group defined for some instances, this function will use empty string as the default.
+   */
+  @Override
+  public synchronized void onInstanceConfigChange(List<InstanceConfig> instanceConfigs, NotificationContext context) {
+    this.instanceConfigs = instanceConfigs;
+    updateInstanceGroupIdMappings();
+  }
+
+  @Override
+  public void onClusterConfigChange(ClusterConfig clusterConfig, NotificationContext context) {
```

Comment:
Shall we make this method to be `synchronized` too? otherwise `faultZoneType` can be changed in the middle of executing `updateInstanceGroupIdMappings`, which can result in inconsistent state. 
Considering the cluster config update rate is very low, the cost of `synchronized` should be minimal or we can make the following section to be `synchronized` to avoid unnecessary locking of other cluster config change:
```
synchronized (this) {
       this.faultZoneType = updatedFaultZoneType;
       updateInstanceGroupIdMappings();
}
```

Reply:
Yeah, that's a good callout